api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +623,50 @@ paths:
           schema:
             $ref: '#/definitions/Problem'
 
+  '/event-types/{name}/cursors':
+      get:
+        tags:
+          - unmanaged-api
+          - monitoring
+          - management-api
+        security:
+          - oauth2: ['nakadi.event_stream.read']
+        description: |
+          Transforms a list of Cursors with `shift` into a list without `shift`s. This is useful when there is the need
+          for randomly access events in the stream.
+
+        parameters:
+          - name: name
+            in: path
+            description: EventType name
+            type: string
+            required: true
+          - name: X-Flow-Id
+            in: header
+            description: |
+              The flow id of the request, which is written into the logs and passed to called services. Helpful
+              for operational troubleshooting and log analysis.
+            type: string
+          - name: X-nakadi-cursors
+            in: header
```

Comment:
@tkrop 

> We will shortly improve our API guidelines, to give better guidance on 
> the usage of headers and get parameters with respect to semantic and 
> size constraints.

@tkrop Understood, the challenge is in the absence of clear guidance, the API owners have to guess what the guidelines might be in advance of them existing (this pattern has gotten through a guild review in the past, so there's no way to have known it's now non-preferred in advance).

With that it mind I see the following options:

- Proceed with the pattern that's here. Add the guild's preferred option in the future on both resources when it exists and deprecate the header pattern (assuming the team is happy to support both options).

- Wait for the pattern to be in place from the guild

- Design a non-header option, hope it lines up with the guild model.

- Have the guild work directly with the team now, and document the outcome.


/+ @ePaul @fmueller as they are often stewards of the guild process :)